src/ape/contracts/base.py

```diff
@@ -899,10 +900,18 @@ def receipt(self) -> ReceiptAPI:
                 return receipt
 
         # Brute force find the receipt.
-        receipt = self.chain_manager.contracts.get_creation_receipt(self.address)
+        receipt = self.meta.receipt
         self._cached_receipt = receipt
         return receipt
 
+    @cached_property
+    def meta(self):
+        """
+        Contract creation details: deployer, factory, block, receipt.
+        """
+        query = ContractCreationQuery(columns=["*"], contract=self.address)
+        return next(self.query_manager.query(query))
```

Comment:
Even if unlikely or impossible on EVM, can we avoid the `StopIteration` here and raise a different error or return None?

Reply:
not possible to hit that branch for an initialized contract instance

Reply:
I was imagining some weird ecosystem that didn't use `EthereumNodeProvider`, even if it was like some sort of Mock-EcosysemAPI, but maybe you're right. No big deal. I just really hate getting random `StopIteration` in tests or anything, it seems like it always happens to me 😅